tests/managedmcg_controller_test.go

```diff
@@ -6,45 +6,189 @@ import (
 		It("should not return validation error", func() {
 			By("using default values", func() {
+				newNamespace := ns.DeepCopy()
 				newManagedMCG := managedMCG.DeepCopy()
-				err := k8sClient.Create(context.Background(), newManagedMCG, &client.CreateOptions{})
+				r.Client = fake.NewClientBuilder().WithScheme(k8sClient.Scheme()).WithObjects(newNamespace, newManagedMCG).Build()
+
+				err := r.Client.Get(context.Background(), client.ObjectKeyFromObject(newManagedMCG), newManagedMCG)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Delete(context.Background(), newManagedMCG, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(newNamespace), newNamespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Delete(context.Background(), newNamespace, &client.DeleteOptions{})
 				Expect(err).NotTo(HaveOccurred())
-				err = k8sClient.Delete(context.Background(), newManagedMCG, &client.DeleteOptions{})
+			})
+		})
+	})
+})
+
+var _ = Describe("ManagedMCGReconciler Reconcile", func() {
+	When("Creating ManagedMCG", func() {
+		newNamespace := ns.DeepCopy()
+		newManagedMCG := managedMCG.DeepCopy()
+
+		addonSecretFake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "addon-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		addonSecretFake.Data = make(map[string][]byte, 1)
+		addonSecretFake.Data["addonparam"] = []byte("foo")
+
+		pagerDutySecretFake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pagerduty-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		pagerDutySecretFake.Data = make(map[string][]byte, 1)
+		pagerDutySecretFake.Data["PAGERDUTY_KEY"] = []byte("foo")
+
+		deadMansSecretfake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "deadmans-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		deadMansSecretfake.Data = make(map[string][]byte, 1)
+		deadMansSecretfake.Data["SNITCH_URL"] = []byte("foo")
+
+		smtpSecretFake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "smtp-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		smtpSecretFake.Data = make(map[string][]byte, 1)
+		smtpSecretFake.Data["host"] = []byte("host")
+		smtpSecretFake.Data["port"] = []byte("8080")
+		smtpSecretFake.Data["username"] = []byte("username")
+		smtpSecretFake.Data["password"] = []byte("password")
+
+		ocscsv := opv1a1.ClusterServiceVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ocs-operator",
+				Namespace: namespace,
+			},
+		}
+
+		noobacsv := noobaav1alpha1.NooBaa{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "nooba-operator",
+				Namespace: namespace,
+			},
+		}
+
+		BeforeEach(func() {
+			r.Scheme = k8sClient.Scheme()
+			r.Client = fake.NewClientBuilder().WithScheme(k8sClient.Scheme()).WithObjects(newNamespace, newManagedMCG, &ocscsv, &smtpSecretFake, &deadMansSecretfake, &pagerDutySecretFake, &noobacsv).Build()
```

Comment:
We don't need all of these resource for **each** test case. Please put only those resource here that are common in all cases, and create the uncommon ones inside their specific cases. This helps make cases more predictable, and sensible in general.